src/elementPlugins/ApplicationViewWrapper.tsx

```diff
@@ -1,30 +1,80 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { ErrorBoundary, pluginProvider } from './'
 import { ApplicationViewWrapperProps, PluginComponents } from './types'
 import evaluateExpression from '@openmsupply/expression-evaluator'
 import { useUpdateResponseMutation } from '../utils/generated/graphql'
+import { ResponseFull } from '../utils/types'
 
 const ApplicationViewWrapper: React.FC<ApplicationViewWrapperProps> = (props) => {
   const {
-    templateElement: { elementTypePluginCode: pluginCode },
+    templateElement,
     isVisible,
     isEditable,
     isRequired,
     currentResponse,
+    allResponses,
   } = props
 
-  if (!pluginCode || !isVisible) return null
+  const { elementTypePluginCode: pluginCode } = templateElement
+  const { validation: validationExpression, validationMessage } = templateElement?.parameters
 
   const [responseMutation] = useUpdateResponseMutation()
+  const [pluginMethods, setPluginMethods] = useState({
+    validate: (validationExpress: any, validationMessage: any, evaluator: any) =>
+      console.log('notLoaded'),
+  })
+  const [validationState, setValidationState] = useState<any>({})
+
+  useEffect(() => {
+    if (!pluginCode) return
+    // TODO use generic or plugin specific
+    setPluginMethods({
+      validate: defaultValidate,
+    })
+  }, [])
+
+  const onUpdate = async (value: string | null | undefined) => {
+    const responses = { thisResponse: value, ...allResponses }
 
-  const onUpdate = (updateObject: any) => {
-    const { isValid, value } = updateObject
+    if (!validationExpression || value === undefined) {
+      setValidationState({ isValid: true })
+      return { isValid: true }
+    }
+
+    const evaluator = async (expression: any) => {
+      return await evaluateExpression(expression, { objects: [responses], APIfetch: fetch })
+    }
+    const validationResult = await pluginMethods.validate(
+      validationExpression,
+      validationMessage,
+      evaluator
+    )
+    setValidationState(validationResult)
+
+    return validationResult
+  }
+
+  const onSave = async (jsonValue: ResponseFull) => {
+    const validationResult: any = await onUpdate(jsonValue.text)
+    console.log('On Save', {
+      variables: {
+        id: currentResponse?.id as number,
```

Comment:
Hmmm since we are doing the mutation here using the response `id` I don't think this can be `null`. Can we change the type to always expect the response to be of type `ApplicationResponse` please

Reply:
After that many places that are using this variable in here (and all plugins `ApplicationView` functions) can stop using the `currentResponse?.`. We can start shaping the variables to be checked before they reach a certain point in the App, and then we don't need to keep checking for `undefined` or nulls

Reply:
I'm not sure I understand. The `currentResponse` is typed as `ApplicationResponse | null` (in types.ts) and it *can* be `null` as all responses start off as `null`, right? Maybe you can show me what you mean?

Reply:
No problem, doesn't need to be done here. I've been working with to reuse more common types between the components (and ApplicationViewWrapper) and I can consider adding this in there.